upgrade/v100to101/upgrade.go

```diff
@@ -51,8 +48,12 @@ func UpgradeInstanceManagerPods(namespace string, lhClient *lhclientset.Clientse
 			return errors.Wrapf(err, upgradeLogPrefix+"failed to find pod for instance manager %v during the instance managers pods upgrade", im.Name)
 		}
 		for _, pod := range imPodsList.Items {
-			if err = upgradeInstanceMangerPodLabel(&pod, &im, kubeClient, namespace); err != nil {
-				return err
+			newInstanceManagerLabels := types.GetInstanceManagerLabels(im.Spec.NodeID, im.Spec.Image, im.Spec.Type)
+			imImageLabelKey := types.GetLonghornLabelKey(types.LonghornLabelInstanceManagerImage)
+			if pod.Labels[imImageLabelKey] != newInstanceManagerLabels[imImageLabelKey] {
+				if err := upgradeInstanceMangerPodLabel(&pod, &im, kubeClient, namespace); err != nil {
```

Comment:
Will `upgradeInstanceMangerPodLabel` respect the existing labels?

Is it possible for `pod.Labels` to be nil?

Reply:
> Is it possible for pod.Labels to be nil?
Fixed. Will update `pod.Labels` if it is nil

> Will upgradeInstanceMangerPodLabel respect the existing labels?

Not sure by reading the code. Testing it on a real cluster now.

Reply:
You can use https://github.com/longhorn/longhorn-manager/blob/master/datastore/longhorn.go#L241 as an example.

Reply:
thank you! fixed

Reply:
Yeah, you are right. It doesn't respect existing label. confirm by test on a cluster as well as this line overwriting old labels:
https://github.com/longhorn/longhorn-manager/blob/2d90012fbf3ac220ce9efe993e89a7f98f0865bb/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L221